packages/twenty-front/src/modules/object-record/record-filter/utils/compute-record-gql-operation-filter/checkIfShouldComputeEmptinessFilter.ts

```diff
@@ -0,0 +1,31 @@
+import { FieldMetadataItem } from '@/object-metadata/types/FieldMetadataItem';
+import { getFilterTypeFromFieldType } from '@/object-metadata/utils/formatFieldMetadataItemsAsFilterDefinitions';
+import { FilterableFieldType } from '@/object-record/record-filter/types/FilterableFieldType';
+import { RecordFilter } from '@/object-record/record-filter/types/RecordFilter';
+import { isEmptinessOperand } from '@/object-record/record-filter/utils/isEmptinessOperand';
+
+export const checkIfShouldComputeEmptinessFilter = ({
+  recordFilter,
+  correspondingFieldMetadataItem,
+}: {
+  recordFilter: RecordFilter;
+  correspondingFieldMetadataItem: Pick<FieldMetadataItem, 'type'>;
+}) => {
+  const isAnEmptinessOperand = isEmptinessOperand(recordFilter.operand);
+
+  const filterTypesThatHaveNoEmptinessOperand: FilterableFieldType[] = [
+    'BOOLEAN',
+  ];
+
+  const filterType = getFilterTypeFromFieldType(
```

Comment:
out of curiousity, why do we need a filterType here, is'nt the fieldType enough? I feel there is a 1:1

Reply:
In practice it's not because we have `FilterableFieldType` which is a string literal and `FieldMetadataType` which is an enum, and some field types are excluded like `TS_VECTOR` and `UUID`.